### PR TITLE
Fixes anomalies not spawning right

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Fixes a bug that prevents anomalies from properly spawning, and causing all sorts of trouble (like trying to delete themselves when they don't really exist).

## Why It's Good For The Game
Less bugs is good.

## Testing
Started some anomaly events, they functioned as intended.

## Changelog
:cl:
fix: Anomalies should now spawn correctly.
/:cl:

### CODE REVIEW

File: code/modules/events/anomaly_event.dm [hunk 1]
```diff
@@ -15,26 +15,27 @@
...
 /datum/event/anomaly/proc/find_targets(warn_on_fail = FALSE)
-	var/turf/target
 	for(var/tries in 0 to TURF_FIND_TRIES)
 		impact_area = findEventArea()
 		if(!impact_area)
 			if(warn_on_fail)
-				CRASH("No valid areas for anomaly found.")
-			else
-				return
+				log_debug("No valid areas for anomaly found.")
```
Comment (Contrabang) [inline code/modules/events/anomaly_event.dm:22]:
```suggestion
				stack_trace("No valid areas for anomaly found.")
```

File: code/modules/events/anomaly_event.dm [hunk 2]
```diff
@@ -15,26 +15,27 @@
...
 			var/turf/candidate = pick_n_take(candidate_turfs)
 			if(!is_blocked_turf(candidate,TRUE))
-				target = candidate
+				target_turf = candidate
 				break
 		if(target_turf)
 			break
 	if(!target_turf)
-		CRASH("Anomaly: Unable to find a valid turf to spawn the anomaly. Last area tried: [impact_area] - [impact_area.type]")
+		log_debug("Anomaly: Unable to find a valid turf to spawn the anomaly. Last area tried: [impact_area] - [impact_area.type]")
```
Comment (Contrabang) [inline code/modules/events/anomaly_event.dm:34]:
```suggestion
		stack_trace("Anomaly: Unable to find a valid turf to spawn the anomaly. Last area tried: [impact_area] - [impact_area.type]")
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
these should probably be still logged as runtimes, just not crash it. I recommend stack_trace

**Comment by Henri215** [comment]
changelog has an extra line

**Review by Henri215 — APPROVED** [review]
Lewcs Good To Me

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**